pandas/core/indexes/datetimelike.py

```diff
@@ -651,6 +652,15 @@ def __add__(self, other):
                 raise TypeError("cannot add {typ1} and {typ2}"
                                 .format(typ1=type(self).__name__,
                                         typ2=type(other).__name__))
+            elif isinstance(other, np.ndarray):
+                if is_timedelta64_dtype(other):
+                    return self._add_delta(TimedeltaIndex(other))
+                elif is_integer_dtype(other):
+                    return NotImplemented
```

Comment:
Do we have a test that hits this branch?

Reply:
Yes but indirectly, without this branch it was breaking a couple of other tests that added a `PeriodIndex` and a `np.array(dtype=int64)`. I didn't check what the behavior is there, but could add a more direct test.

Reply:
Direct test is certainly appreciated.  Go for it!

Reply:
Okay, I looked into the cases getting triggered with `PeriodIndex` and unfortunately they're buried pretty deeply in internals (e.g. not obvious this would get called from any public API, at least any more so than the tests that were already reaching this). Also @jreback commented that this should only be internal facing, so I figure its better not to "advertise" this API in a test.